### PR TITLE
Update README.md to include details about dependency injection

### LINKED ISSUE #1773: fo-dicom v5 Dependency Injection Documentation
To make Fellow Oak DICOM use Microsoft.Extensions.Logging, you need to use dependency injection.

If you are using the host builder from ASP.NET Core, you can do this:

```csharp
var host = Host.CreateDefaultBuilder(args)
    .ConfigureLogging(logging =>
    {
        logging.ClearProviders();
        logging.AddConsole();
        logging.SetMinimumLevel(LogLevel.Information);
    })
    .ConfigureServices(services =>
    {
        services.AddFellowOakDicom();
        services.AddHostedService<Worker>();
        // This will be injected into the DICOM service
        services.AddSingleton<CustomDependency>();
    })
    .Build();

// This is still necessary for now, but we want to get rid of this in the long term
DicomSetupBuilder.UseServiceProvider(host.Services);

host.Run();
```

If you are not using the host builder, you'll need to make your own service collection:

```csharp
var services = new ServiceCollection();
services.AddLogging(logging =>
{
    logging.ClearProviders();
    logging.AddConsole();
    logging.SetMinimumLevel(LogLevel.Information);
});
services.AddFellowOakDicom();
var serviceProvider = services.BuildServiceProvider();
DicomSetupBuilder.UseServiceProvider(serviceProvider);
```

_Originally posted by @amoerie in https://github.com/fo-dicom/fo-dicom/discussions/1627#discussioncomment-6439203_

The above post is gold - and is the first time I've seen a clear example of using DI with fo-dicom v5. Dependency injection documentation, samples would be very useful

### PR DESCRIPTION
**Pull request by amoerie**
Add documentation about dependency injection support

Fixes #1773

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- ~~I have updated API documentation~~
- ~~I have included unit tests~~
- ~~I have updated the change log~~
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Update readme with clarifications about dependency injection support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,7 +102,20 @@ new DicomSetupBuilder()
 #### Dependency injection support
 
 Whenever you use APIs of Fellow Oak DICOM such as `DicomFile.Open`, `DicomServerFactory.Create`, the global statically registered service provider (`DicomSetupBuilder.UseServiceProvider`) will be used to resolve dependencies.  
-It is also possible to inject custom dependencies into your DICOM services, but with one important constraint: **you must have these exact three constructor parameters: INetworkStream stream, Encoding fallbackEncoding, ILogger logger**. (The names and their order don't matter, but the types do) 
+Please note that using dependency injection is generally preferred over the static APIs, if they are available.
```
Comment (mrbean-bremen) [inline README.md]: "if they are available" - you mean dependency injection? In this case "if it is available"

File: README.md [hunk 2]
```diff
@@ -104,7 +102,20 @@ new DicomSetupBuilder()
...
+
+| Use case                                      | Static API                                                | Can use dependency injection ?                   |
+| --------------------------------------------- | --------------------------------------------------------- | ------------------------------------------------ |
+| Creating a DICOM server                       | DicomServerFactory.Create                                 | Yes, use `IDicomServerFactory`                   |
+| Creating a DICOM client                       | DicomClientFactory.Create                                 | Yes, use `IDicomClientFactory`                   |
+| Creating an advanced DICOM client connection  | AdvancedDicomClientConnectionFactory.OpenConnectionAsync  | Yes, use `IAdvancedDicomClientConnectionFactory` |
+| Opening a DICOM file                          | DicomFile.Open(..)                                        | No                                               |
```
Comment (mrbean-bremen) [inline README.md]: Maybe add `DicomFile.OpenAsync(..)`

File: README.md [hunk 3]
```diff
@@ -104,7 +102,20 @@ new DicomSetupBuilder()
...
+| Rendering a DICOM file                        | new DicomImage(..).RenderImage(..)                        | No                                               |
+
+#### Injecting custom dependencies into DICOM services
+
+It is possible to inject custom dependencies into your DICOM services, but with one important requirement: **you must have these exact three constructor parameters: INetworkStream stream, Encoding fallbackEncoding, ILogger logger**. 
+The names and their order don't matter, but the types do.
 Yes, `ILogger` is a non-generic typed logger. If you want a `Logger<T>`, you can **add** an extra constructor parameter and use it however you see fit.
 
 Here is a fully standalone working example:
```
Comment (mrbean-bremen) [inline README.md]: "full standalone working example" or "fully working standalone example" (probably the latter is meant)

File: README.md (hunk 1 above)
Comment (amoerie) [reply README.md]: Changing to "if it is available", thanks

File: README.md (hunk 2 above)
Comment (amoerie) [reply README.md]: I'll just use DicomFile.OpenAsync instead, it's probably what you should be using 90% of the time anyway.

File: README.md (hunk 3 above)
Comment (amoerie) [reply README.md]: Changed to "full standalone example", sounds OK?

**Review by mrbean-bremen — APPROVED** [review]

**Review by gofal — APPROVED** [review]

**Pull request merged**